Source/ASDisplayNode.mm

```diff
@@ -729,6 +729,8 @@ - (BOOL)_isNodeLoaded
 
 - (UIView *)view
```

Comment:
Per the documentation [here](https://github.com/facebook/AsyncDisplayKit/blob/master/Source/ASDisplayNode.h#L202), I think these getters can be called on a background thread (after the first time), given that the returned view/layer is not used on that thread. I can't think of any use case in which calling them off main is needed, but I also don't want to break the API contract.

Can we instead opt for acquiring the lock in these methods?